controller/dataFetch.js

```diff
@@ -102,18 +102,31 @@ dataFetchManager.getMetrics = async function (metricIds, ticker) {
 
 }
 
-dataFetchManager.getQuotes = async function (timeRange, ticker) {
+dataFetchManager.getQuotes = function (timeRange, ticker) {
     // get data from iex server
-    if (timeRange !== undefined && timeRange.length > 0) {
-        let response = await iexRequest.test.historicalPrices()
-        dataObject.daily.historicalPrices = response.data;
-        return dataObject.daily.historicalPrices;
+    if (timeRange !== undefined && timeRange.length > 0 && timeRange === "1d") {
+        let response = iexRequest.test.historicalPrices1d()
+        return response;
+    }
+    if (timeRange !== undefined && timeRange.length > 0 && timeRange === "1m") {
+        let response = iexRequest.test.historicalPrices1m()
+        return response;
+    }
+    if (timeRange !== undefined && timeRange.length > 0 && timeRange === "3m") {
+        let response = iexRequest.test.historicalPrices3m()
+        return response;
+    }
+    if (timeRange !== undefined && timeRange.length > 0 && timeRange === "1y") {
+        let response = iexRequest.test.historicalPrices1y()
+        return response;
+    }
+    if (timeRange !== undefined && timeRange.length > 0 && timeRange === "5y") {
+        let response = iexRequest.test.historicalPrices5y()
+        return response;
```

Comment:
@mariarabin Could you update the iexRequest function to take in period/timerange parameter and return a response? So you don't have 5 identical sets of code, which will be hard to maintain.

Reply:
Let me know if you have questions